python/graphstorm/trainer/gsgnn_trainer.py

```diff
@@ -89,15 +88,23 @@ def setup_task_tracker(self, task_tracker):
         task_tracker : GSTaskTracker
             The task tracker
         """
-        if self.evaluator is not None:
-            self.evaluator.setup_task_tracker(task_tracker)
         self._task_tracker = task_tracker
 
     def setup_evaluator(self, evaluator):
-        """ Set the evaluator
+        """ Setup the evaluator
+
+        If the evaluator has its own task tracker, just setup the evaluator. But if the evaluator
+        has no task tracker, will use this Trainer's task tracker to setup the evaluator. When there
+        is no self task tracker, will create a new one by using the given evaluator's evaluation
+        frequency.
         """
-        if self.task_tracker is not None:
-            evaluator.setup_task_tracker(self.task_tracker)
+        if evaluator.task_tracker is None:
+            if self.task_tracker is not None:
+                task_tracker = GSSageMakerTaskTracker(evaluator.eval_frequency)
+                evaluator.setup_task_tracker(task_tracker)
+            else:
+                evaluator.setup_task_tracker(self.task_tracker)
```

Comment:
ditto

Reply:
The self.task_tracker belongs to the trainer, and setup_task_tracker will need this argument.